### PR TITLE
Make region name or self-link acceptable for bound_region

### LINKED ISSUE #33: bound_region is fully qualified URL, bound_zone is not
I genuinely don't know if this is working as intended or not.

I expected `bound_region` to support something like `europe-west4`. But it appears that the region actually has to be a fully qualified URL like: `https://www.googleapis.com/compute/v1/projects/my-project/regions/europe-west4`. Problem one I ran into is that isn't documented anywhere I found obvious... 

So that would be fine, but then I realized `bound_zone` _does accept_ `europe-west4-2` and _does not_ use fully qualified URLs. So this is a bit of a weird UX because two very similar bindings require two different formats for values and this isn't made clear.

I don't know if this is correct or not, but at best its a weird UX that we should probably document. Thanks!

### PR DESCRIPTION
**Pull request by emilymye**
Fix #33.

### CODE REVIEW

**Comment by emilymye** [comment]
Added docs and decided we should probably also allow self-link for zone.

**Comment by emilymye** [comment]
@jefferai @briankassouf PTAL again when you get the chance

File: plugin/path_login.go [hunk 1]
```diff
@@ -584,6 +573,20 @@ func (b *GcpAuthBackend) authorizeGCEInstance(ctx context.Context, instance *com
 	return nil
 }
 
+func compareResourceNameOrSelfLink(expected, actual, collectionId string) bool {
+	sep := fmt.Sprintf("%s/", collectionId)
+	if strings.Contains(expected, sep) {
+		return expected == actual
+	}
+
+	expTkns := strings.SplitAfter(expected, sep)
```
Comment (kalafut) [inline plugin/path_login.go]: Are these two lines necessary? `expected` won't contain `sep` at this point, so I think `expectedName` is always the same as `expected`?

Comment (emilymye) [reply plugin/path_login.go]: fair

**Comment by jefferai** [comment]
Looks good... 👍 to merge?

**Comment by emilymye** [comment]
Please!

**Pull request merged**